public/maillog.php

```diff
@@ -0,0 +1,150 @@
+<?php
+/*
+* to do :   - add pagination
+*	          - sort by mounth
+* 	        - remove old logs
+*/
+
+require_once('common.php');
+
+authentication_require_role('admin');
+
+$CONF = Config::getInstance()->getAll();
+$smarty = PFASmarty::getInstance();
+
+$PALANG = $CONF['__LANG'];
+
+$SESSID_USERNAME = authentication_get_username();
+if (authentication_has_role('global-admin')) {
+    $list_domains = list_domains();
+} else {
+    $list_domains = list_domains_for_admin($SESSID_USERNAME);
+}
+
+
+/*
+foreach ($list_domains as $domain){
+	echo $domain. '<br>';
+}
+*/
+
+$fDomain=$list_domains[0];
+
+if ($_SERVER['REQUEST_METHOD'] == "GET") {
+    if (isset($_GET['fDomain'])) {
+        $fDomain_aux = escape_string($_GET['fDomain']);
+        $flag_fDomain = 0;
+	//check if domain exists
+        if ((is_array($list_domains) and sizeof($list_domains) > 0)) {
```

Comment:
`$list_domains` will always be an array (unless `list_domains()` or `list_domains_for_admin()` fails - but you should check for such an error earlier.

Also, the `sizeof()` check looks a bit superfluous - with an empty array, the foreach loop will simply run 0 times.

Speaking about the foreach loop in the next lines - unless I missed something, this looks like a manual implementation of https://www.php.net/in_array